### PR TITLE
Fix file permissions set on token store

### LINKED ISSUE #2: Default permissions on token_store are not secure
When writing new `token_store` to `DEFAULT_TOKEN_FILE` likely here...

https://github.com/globus/globus-timer-cli/blob/58633637e4bdb19995abc5237ecc8546a6f19278/timer_cli/auth.py#L104-L119

...the permissions of the `DEFAULT_TOKEN_FILE` file **are not secure**.

The default behavior for the creation of the `DEFAULT_TOKEN_FILE` should be to create it with octal mode `0o600`.

For example...
```
original_umask = os.umask(0o177)  # 0o777 ^ 0o600
try:
    with os.fdopen(os.open(filename, os.O_WRONLY | os.O_CREAT, 0o600), 'w') as fp:
        json.dump(tokens, fp, indent=4, separators=(", ", ": "))
finally:
    os.umask(original_umask)
```

### PR DESCRIPTION
**Pull request by rudyardrichter**
Change `TokenCache` to set permissions to `0o600` for the token store file, and
also check on initializtion that the specified file has the correct permissions,
fixing it if needed.

Resolves #2.

### CODE REVIEW

File: timer_cli/auth.py [hunk 1]
```diff
@@ -108,14 +116,20 @@ def default(x):
...
-            with open(self.token_store, "w") as f:
-                json.dump(
-                    {k: v._asdict() for k, v in self.tokens.items()},
-                    f,
-                    indent=2,
-                    sort_keys=True,
-                    default=default,
-                )
+            # disable permissions other than user read/write
+            original_umask = os.umask(0o177)
```
Comment (urimandujano) [inline timer_cli/auth.py:120]: I'm not familiar with the octals - what does temporarily setting the umask to 0o177 do for the subsequent file operations?

**Review by urimandujano — APPROVED** [review]
From my local tests it looks like the token cache is correctly getting set to owner readable/writable only:

Before
```
❯ stat .globus_timer_tokens.json
16777220 17436036 -rw-r--r-- 1 uriel staff ...
```
After
```
❯ stat ~/.globus_timer_tokens.json
16777220 17436036 -rw------- 1 uriel staff ...
```

File: timer_cli/auth.py (hunk 1 above)
Comment (rudyardrichter) [reply timer_cli/auth.py:120]: The `umask` setting _disables_ permissions for files created in a process, so a setting of 177 here means that the execute permission for users will be disabled and all permissions for anyone else are disabled.

Comment (urimandujano) [reply timer_cli/auth.py:120]: Gotcha! Thanks for the explanation.

**Pull request merged**